expertsystem/data.py

```diff
@@ -164,16 +147,27 @@ def mass(self) -> float:
     def width(self) -> float:
         return self.__energy.imag
 
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, ComplexEnergy):
+            return self.complex_energy == other.complex_energy
+        raise NotImplementedError
 
-class ComplexEnergyState(HasComplexEnergy):
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}{self.complex_energy}"
+
+
+class ComplexEnergyState(ComplexEnergy):
     """Pole in the complex energy plane, with quantum numbers."""
 
-    def __init__(self, energy: complex, state: QuantumState):
+    def __init__(self, energy: complex, state: QuantumState[Spin]):
         super().__init__(energy)
-        self.state: QuantumState = state
+        self.state: QuantumState[Spin] = state
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}{self.complex_energy, self.state}"
 
 
-class Particle(HasComplexEnergy):
+class Particle(ComplexEnergy):
```

Comment:
Naming, as mentioned before: "`Particle` is `ComplexEnergy`" would sound confusing.

Reply:
That would also be resolved with the decorator